hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/clustering/ClusteringCommitSink.java

```diff
@@ -119,7 +119,16 @@ private void commitIfNecessary(String instant, List<ClusteringCommitEvent> event
       return;
     }
 
-    if (events.stream().anyMatch(ClusteringCommitEvent::isFailed)) {
+    // here we should take the write errors under consideration
+    // as some write errors might cause data loss when clustering
+    List<WriteStatus> statuses = events.stream()
```

Comment:
Things are a little different for when `events.stream().anyMatch(ClusteringCommitEvent::isFailed)`, the `isFailed` flag always indicates some errors that are related with the service execution, not the data record quality, we should always try to rollback in this case, not just throws.

Reply:
Agree that `isFailed` indicates the execution failure always to be rollbacked. 

So in the updated we will judge whether to rollback write status errors when the config `FlinkOptions.IGNORE_FAILED` false.

Pls take a review.